### DISCUSSION TITLE
General queries (Obsidian Dataview)

### DISCUSSION THREAD

**Discussion by srid** [srid/emanote#106] [Ideas]
Basically do this: https://github.com/blacksmithgu/obsidian-dataview#obsidian-dataview

Examples: https://forum.obsidian.md/t/dataview-plugin-snippet-showcase/13673

## Specific use cases

- #141 
- #139

**Comment by srid** [comment]
@j-steinbach I've added two of your recent ideas under this bucket (see the edited description above). If you have any more relevant query ideas, feel free to add them here as a comment.

**Comment by j-steinbach** [comment]
Turn queries into buttons.

In https://github.com/srid/emanote/discussions/139# we talked about how queries are "local", i.e. they get shown on the current note. And in (https://github.com/srid/emanote/discussions/141) I talked about "special views" for queries, i.e. the timeline.

Why not combine these into query + action?

For example I could add a menu button that executes the "select a random note" query *and then navigates to it*. The query gets combined with an action: navigate! (https://github.com/srid/emanote/discussions/139#discussioncomment-1370702)

Same with the timeline: Query for notes tagged "event", then navigate to a specific time-line template and display them there.

This would allow extension of the UI. For example I could add a button to view today's daily note. Or navigate to the most recent project note. Or the the last added note. 

TLDR: Custom buttons in the menu that contain a pre-defined query and navigate to the corresponding site.

**Comment by j-steinbach** [comment]
Not sure if we already discussed that, but I think it would be cool if we could get an overview over today's notes. Some sort of aggregation.

![emanote-timeline-query](https://user-images.githubusercontent.com/69524139/143685041-5c5ac00e-39d9-42de-a198-2b0aa8209abf.png)

**Reply by srid** [reply]
Tangential solution: https://github.com/srid/emanote/discussions/141#discussioncomment-2406971

**Reply by srid** [reply]
I'll probably create an `emanote-with-dailynotes` app that uses Emanote and adds some special views for diary purposes like this. Then later at some point in future, we can do this in Emanote itself in _general_ fashion without hardcoding any purposes.

**Comment by srid** [comment]
We should just support Bases instead https://help.obsidian.md/bases

https://github.com/srid/emanote/issues/601
